src/Catalog.Api/Mappers/CategoryMapper.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Catalog.Api.Requests;
+using Catalog.Api.Responses;
+
+namespace Catalog.Api.Mappers
+{
+    public static class CategoryMapper
+    {
+        public static Domain.Entities.Category Map(CategoryRequest request, int? categoryId = null)
+        {
+            return new Domain.Entities.Category
+            {
+                Id = categoryId ?? 0,
+                Name = request.Name,
+                ImageUrl = request.ImageUrl,
+                Parent = request.ParentCategoryId != null ? new Domain.Entities.Category
+                {
+                    Id = (int)request.ParentCategoryId
+                } : null
+            };
+        }
+
+        public static CategoryResponse Map(Domain.Entities.Category category)
+        {
+            var response = new CategoryResponse
+            {
+                Id = category.Id,
+                Name = category.Name,
+                ImageUrl = category.ImageUrl,
+                ParentCategoryId = category.Parent?.Id,
+                Links = new List<Link>
+                {
+                    new Link
+                    {
+                        Href = $"/catalog/category/{category.Id}",
+                        Rel = "self",
+                        Method = "GET"
+                    },
+                    new Link
+                    {
+                        Href = $"/catalog/category/{category.Id}",
+                        Rel = "update_category",
+                        Method = "PUT"
+                    },
+                    new Link
+                    {
+                        Href = $"/catalog/category/{category.Id}",
+                        Rel = "delete_category",
+                        Method = "DELETE"
+                    },
+                    new Link
+                    {
+                        Href = $"/catalog/item/{category.Id}{{?pageSize,page}}",
```

Comment:
`{{?pageSize,page}}` what does this code do?

Reply:
That is the link in the response for the GET endpoint, and it was suggested to show the available query parameters like this
https://stackoverflow.com/questions/6022230/how-can-i-use-hateoas-and-query-parameters-for-restful-search